src/scatter_buildblock/ScatterEstimationByBin.cxx

```diff
@@ -11,447 +11,1274 @@
+        }
+
+        if(ffw_project_mask_image() == Succeeded::no)
+        {
+            warning("Unsuccessfull to fwd project the mask image. Abord.");
+            return Succeeded::no;
+        }
+    }
+
+    info(">>>>Set up finished successfully!!<<<<");
+    return Succeeded::yes;
 }
 
-void
+Succeeded
 ScatterEstimationByBin::
-set_density_image(const std::string& filename)  
+set_up_iterative(IterativeReconstruction<DiscretisedDensity<3, float> > * iterative_object)
 {
-  this->density_image_filename=filename;
-  this->density_image_sptr= 
-    read_from_file<DiscretisedDensity<3,float> >(filename);
-  if (is_null_ptr(this->density_image_sptr))
+    info("Setting up iterative reconstruction ...");
+    iterative_object->set_input_data(this->input_projdata_2d_sptr);
+
+    //
+    // Multiplicative projdata
+    //
+
+    shared_ptr<BinNormalisation> attenuation_correction(new TrivialBinNormalisation());
+    shared_ptr<BinNormalisation> normalisation_coeffs(new TrivialBinNormalisation());
+
+    if (is_null_ptr(this->atten_projdata_sptr))
     {
-      error(boost::format("Error reading density image %s") %
-            this->density_image_filename);
+        warning("No attenuation projdata have been initialised. "
+                "Forward Projecting the attenuation image... ");
+
+        //Project the attenuation imge.
+        shared_ptr<ForwardProjectorByBin> forw_projector_sptr;
+        shared_ptr<ProjMatrixByBin> PM(new  ProjMatrixByBinUsingRayTracing());
+        forw_projector_sptr.reset(new ForwardProjectorByBinUsingProjMatrixByBin(PM));
+        info(boost::format("\n\nForward projector used for the calculation of\n"
+                           "attenuation coefficients: %1%\n") % forw_projector_sptr->parameter_info());
+        forw_projector_sptr->set_up(this->input_projdata_sptr->get_proj_data_info_ptr()->create_shared_clone(),
+                                    this->atten_image_sptr );
+
+        // I think I could project directly to 2D ...
+        // - No, I would still need the 3D version at the very end.
+        if (this->atten_coeff_filename.size() > 0)
+            this->atten_projdata_sptr.reset(new ProjDataInterfile(this->input_projdata_sptr->get_exam_info_sptr(),
+                                                                  this->input_projdata_sptr->get_proj_data_info_sptr()->create_shared_clone(),
+                                                                  this->atten_coeff_filename,
+                                                                  std::ios::in | std::ios::out | std::ios::trunc));
+        else
+            this->atten_projdata_sptr.reset(new ProjDataInMemory(this->input_projdata_sptr->get_exam_info_sptr(),
+                                                                 this->input_projdata_sptr->get_proj_data_info_ptr()->create_shared_clone()));
+        forw_projector_sptr->forward_project(* this->atten_projdata_sptr, *this->atten_image_sptr);
```

Comment:
BUG?

this would need the exponential. Far better to use BinNormalisationFromAttenuationImage